Mathlib/RingTheory/Ideal/AssociatedPrime/Finiteness.lean

```diff
@@ -0,0 +1,205 @@
+/-
+Copyright (c) 2025 Jinzhao Pan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jinzhao Pan
+-/
+import Mathlib.LinearAlgebra.Isomorphisms
+import Mathlib.Order.RelSeries
+import Mathlib.RingTheory.Ideal.AssociatedPrime.Basic
+import Mathlib.RingTheory.Noetherian.Basic
+import Mathlib.RingTheory.Spectrum.Prime.Defs
+
+/-!
+
+# Finitely generated module over noetherian ring have finitely many associated primes.
+
+In this file we proved that any finitely generated module over a noetherian ring have finitely many
+associated primes.
+
+## Main results
+
+* `IsNoetherianRing.exists_relSeries_isQuotientEquivQuotientPrime`: If `A` is a Noetherian ring
+  and `M` is a finitely generated `A`-module, then there exists a chain of submodules
+  `0 = M₀ ≤ M₁ ≤ M₂ ≤ ... ≤ Mₙ = M` of `M`, such that for each `0 ≤ i < n`,
+  `Mᵢ₊₁ / Mᵢ` is isomorphic to `A / pᵢ` for some prime ideal `pᵢ` of `A`.
+
+* `IsNoetherianRing.induction_on_isQuotientEquivQuotientPrime`: If a property on
+  finitely generated modules over a Noetherian ring satisfies that:
+
+  - it holds for zero module,
+  - it holds for any module isomorphic to some `A ⧸ p` where `p` is a prime ideal of `A`,
+  - it is stable by short exact sequences,
+
+  then the property holds for every finitely generated modules.
+
+* `associatedPrimes.finite`: There are only finitely many associated primes of a
+  finitely generated module over a Noetherian ring.
+
+-/
+
+universe u v
+
+variable {A : Type u} [CommRing A] [IsNoetherianRing A]
+  {M : Type v} [AddCommGroup M] [Module A M] [Module.Finite A M] {n : ℕ} {N : Submodule A M}
+
+section
+
+omit [Module.Finite A M]
+
+/-- A `Prop` asserting that two submodules `N₁, N₂` satisfy `N₁ ≤ N₂` and
+`N₂ / N₁` is isomorphic to `A / p` for some prime ideal `p` of `A`. -/
+def Submodule.IsQuotientEquivQuotientPrime (N₁ N₂ : Submodule A M) :=
+  N₁ ≤ N₂ ∧ ∃ (p : PrimeSpectrum A), Nonempty ((↥N₂ ⧸ N₁.submoduleOf N₂) ≃ₗ[A] A ⧸ p.1)
+
+private theorem aux (h : N ≠ ⊤) :
```

Comment:
Even if those are private, can you choose a reasonable name? Or at least put them in a namespace.

Reply:
The proof was changed and these private things were removed.